scylla/src/transport/session.rs

```diff
@@ -434,26 +434,30 @@ impl Session {
     /// # }
     /// ```
     pub async fn connect(mut config: SessionConfig) -> Result<Session, NewSessionError> {
+        let known_nodes = std::mem::take(&mut config.known_nodes);
```

Comment:
That's not a good idea. You are moving `known_nodes` out of `SessionConfig`, therefore leaving the config object in a weird state, but valid from the POV of the language. You are actually using `config.known_nodes` later, incorrectly:

```rust
let mut node_addresses: Vec<SocketAddr> = Vec::with_capacity(config.known_nodes.len());
```

The `Vec::with_capacity` call is basically useless because the length is always zero.

In order to prevent such issues I think it is better to just move the `known_nodes` out of `config` without replacing it with a valid value. Unless I'm missing something, Rust should support this - if you move one field out then you can still use other fields of the struct, even though the struct as a whole is not valid and you cannot pass it somewhere else / call its methods.

Reply:
The reason why I decided to act in this manner is being able to call methods `take_pool_config()` and `take_connection_config()`. As they are called only by `connect()`, however, I can simply inline them into there.